test_po_update failure on conifg_reload

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/sonic-net/SONiC/wiki#report-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- SONiC Support Forums - https://groups.google.com/forum/#!forum/sonicproject

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**

<!--
Briefly describe the problem you are having in a few paragraphs.
-->

During the config_reload from test_po_update, seeing following error:

args = (&lt;MultiAsicSonicHost sfd-vt2-lc0&gt;,)
kwargs = {&apos;ignore_loganalyzer&apos;: None, &apos;safe_reload&apos;: True}, loganalyzer = None

    @wraps(func)
    def decorated(*args, **kwargs):
        &quot;&quot;&quot;
        try to fetch loganalyzer instances from kwargs:
        if ignore_loganalyzer is not passed, do nothing but execute the decorated function.
        if ignore_loganalyzer is passed, to avoid &apos;unexpected keyword argument error&apos;,
            delete the ignore_loganalyzer from kwargs so that it would not be passed to the decorated function,
            and set ignore_loganalyzer markers before and after the decorated function on all log analyzer instances.
        &quot;&quot;&quot;

        loganalyzer = None
        if &apos;ignore_loganalyzer&apos; in kwargs and kwargs[&apos;ignore_loganalyzer&apos;] is not None:
            loganalyzer = kwargs[&apos;ignore_loganalyzer&apos;]
            kwargs.pop(&apos;ignore_loganalyzer&apos;)

        if loganalyzer:
            for _, dut_loganalyzer in loganalyzer.items():
                dut_loganalyzer.add_start_ignore_mark()

&gt;       res = func(*args, **kwargs)
E       TypeError: config_reload() got an unexpected keyword argument &apos;ignore_loganalyzer&apos;

args       = (&lt;MultiAsicSonicHost sfd-vt2-lc0&gt;,)
func       = &lt;function config_reload at 0x7fd9c2a4c4d0&gt;
kwargs     = {&apos;ignore_loganalyzer&apos;: None, &apos;safe_reload&apos;: True}
loganalyzer = None


**Steps to reproduce the issue:**
1.
2.
3.

**Describe the results you received:**


**Describe the results you expected:**


**Additional information you deem important:**
<!--
software/ASIC/Hardware Flatform version and info
-->
    **Output of `show version`:**
Using sonic-mgmt 202205 sha 15d9388

    ```
    (paste your output here)
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

I still see this error  when disabled loganalyzer:

&gt;       res = func(*args, **kwargs)
E       TypeError: config_reload() got an unexpected keyword argument &apos;ignore_loganalyzer&apos;

This happens if this test case fails and config reloads is done by test script to recover:
ERROR pc/test_po_update.py::test_po_update_io_no_loss       <<<< This is a different issue being debugged but config reload should work.

--- a/tests/common/config_reload.py
+++ b/tests/common/config_reload.py
@@ -3,7 +3,7 @@ import logging
 import os
 
 from tests.common.helpers.assertions import pytest_assert
-from tests.common.plugins.loganalyzer.utils import ignore_loganalyzer
+from tests.common.plugins.loganalyzer.utils import support_ignore_loganalyzer
 from tests.common.platform.processes_utils import wait_critical_processes
 from tests.common.utilities import wait_until
 from tests.common.configlet.utils import chk_for_pfc_wd
@@ -112,7 +112,7 @@ def pfcwd_feature_enabled(duthost):
     return pfc_status == 'enable' and switch_role not in ['MgmtToRRouter', 'BmcMgmtToRRouter']
 
 
-@ignore_loganalyzer
+@support_ignore_loganalyzer
 def config_reload(sonic_host, config_source='config_db', wait=120, start_bgp=True, start_dynamic_buffer=True,
                   safe_reload=False, wait_before_force_reload=0, wait_for_bgp=False,
                   check_intf_up_ports=False, traffic_shift_away=False, override_config=False,

--- a/tests/common/plugins/loganalyzer/loganalyzer.py
+++ b/tests/common/plugins/loganalyzer/loganalyzer.py
@@ -267,6 +267,10 @@ class LogAnalyzer:
         """
         Adds the start ignore marker to the log files
         """
+        # We copy 'loganalyzer.py' to /tmp dir during loganalyzer initialization,
+        # but the file could be auto removed by rebooting device,
+        # always copy script to make sure the marker can be added successfully.
+        self.ansible_host.copy(src=ANSIBLE_LOGANALYZER_MODULE, dest=os.path.join(self.dut_run_dir, "loganalyzer.py"))
         add_start_ignore_mark = ".".join((self.marker_prefix, time.strftime("%Y-%m-%d-%H:%M:%S", time.gmtime())))
         cmd = "python {run_dir}/loganalyzer.py --action add_start_ignore_mark --run_id {add_start_ignore_mark}"\
             .format(run_dir=self.dut_run_dir, add_start_ignore_mark=add_start_ignore_mark)
@@ -281,6 +285,10 @@ class LogAnalyzer:
         """
         Adds the end ignore marker to the log files
         """
+        # We copy 'loganalyzer.py' to /tmp dir during loganalyzer initialization,
+        # but the file could be auto removed by rebooting device,
+        # always copy script to make sure the marker can be added successfully.
+        self.ansible_host.copy(src=ANSIBLE_LOGANALYZER_MODULE, dest=os.path.join(self.dut_run_dir, "loganalyzer.py"))
         marker = self._markers.pop()
         cmd = "python {run_dir}/loganalyzer.py --action add_end_ignore_mark --run_id {marker}"\
             .format(run_dir=self.dut_run_dir, marker=marker)

--- a/tests/common/plugins/loganalyzer/utils.py
+++ b/tests/common/plugins/loganalyzer/utils.py
@@ -12,10 +12,10 @@ def ignore_loganalyzer(func):
             and set ignore_loganalyzer markers before and after the decorated function on all log analyzer instances.
         """
 
-        loganalyzer = None
-        if 'ignore_loganalyzer' in kwargs and kwargs['ignore_loganalyzer'] is not None:
-            loganalyzer = kwargs['ignore_loganalyzer']
-            kwargs.pop('ignore_loganalyzer')
+        # Need to remove parameter 'ignore_loganalyzer' from kwargs
+        # Otherwise it breaks the decorated func
+        # Since the parameter 'ignore_loganalyzer' is not defined in the signature
+        loganalyzer = kwargs.pop('ignore_loganalyzer', {})
 
         if loganalyzer:
             for _, dut_loganalyzer in list(loganalyzer.items()):

--- a/tests/common/plugins/loganalyzer/utils.py
+++ b/tests/common/plugins/loganalyzer/utils.py
@@ -1,7 +1,7 @@
 from functools import wraps
 
 
-def ignore_loganalyzer(func):
+def support_ignore_loganalyzer(func):
     @wraps(func)
     def decorated(*args, **kwargs):
         """

--- a/tests/common/reboot.py
+++ b/tests/common/reboot.py
@@ -10,6 +10,7 @@ from collections import deque
 from .helpers.assertions import pytest_assert
 from .platform.interface_utils import check_interface_status_of_up_ports
 from .platform.processes_utils import wait_critical_processes
+from .plugins.loganalyzer.utils import ignore_loganalyzer
 from .utilities import wait_until, get_plt_reboot_ctrl
 from tests.common.helpers.dut_utils import ignore_t2_syslog_msgs, create_duthost_console, creds_on_dut
 from tests.common.fixtures.conn_graph_facts import get_graph_facts
@@ -223,6 +224,7 @@ def perform_reboot(duthost, pool, reboot_command, reboot_helper=None, reboot_kwa
     return [reboot_res, dut_datetime]
 
 
+@ignore_loganalyzer
 def reboot(duthost, localhost, reboot_type='cold', delay=10,
            timeout=0, wait=0, wait_for_ssh=True, wait_warmboot_finalizer=False, warmboot_finalizer_timeout=0,
            reboot_helper=None, reboot_kwargs=None, return_after_reconnect=False,

--- a/tests/common/reboot.py
+++ b/tests/common/reboot.py
@@ -10,7 +10,7 @@ from collections import deque
 from .helpers.assertions import pytest_assert
 from .platform.interface_utils import check_interface_status_of_up_ports
 from .platform.processes_utils import wait_critical_processes
-from .plugins.loganalyzer.utils import ignore_loganalyzer
+from .plugins.loganalyzer.utils import support_ignore_loganalyzer
 from .utilities import wait_until, get_plt_reboot_ctrl
 from tests.common.helpers.dut_utils import ignore_t2_syslog_msgs, create_duthost_console, creds_on_dut
 from tests.common.fixtures.conn_graph_facts import get_graph_facts
@@ -224,7 +224,7 @@ def perform_reboot(duthost, pool, reboot_command, reboot_helper=None, reboot_kwa
     return [reboot_res, dut_datetime]
 
 
-@ignore_loganalyzer
+@support_ignore_loganalyzer
 def reboot(duthost, localhost, reboot_type='cold', delay=10,
            timeout=0, wait=0, wait_for_ssh=True, wait_warmboot_finalizer=False, warmboot_finalizer_timeout=0,
            reboot_helper=None, reboot_kwargs=None, return_after_reconnect=False,

